### DISCUSSION TITLE
Creation of a new user without SMTP

### DISCUSSION THREAD

**Discussion by Pyr0technicien** [monarc-project/MonarcAppFO#426] [Q&A] [answered]
Hey, 

We are trying to create a new user on one of our old instance on monarc (adminsitrator don't have the possibility to create a temporary password).
This instance isn't linked to any SMTP Server, i'm trying to figure out how the user were created initially ? is there a default password ?
It's the instance of one of our client, they created the new user, but the "forgot my password" isn't working since it's not linked to any SMTP.

Also, the previous people in charge of monarc left, so I'm trying to understand how I can get this new user to use the account they created ?

I have access to the machine on wich monarc is installed, is there a way to do this directly with a script ?

Thanks

What I have been trying :

- Using the script to create the admin user (from installation procedure). However it tries to create a user with an email already existing, an old employee)
- changing that old employee email and relaunch the script. However password "admin" isn't correct I guess it's due to the seed.
- Copying a hash from an other instance (with the password being known) and setting it in the DB for that new "admin" user created by the script but it seems like password isn't accepted again.

**Comment by ruslanbaidan** [comment]
Hello,

The easiest way of resolving the situation is to upgrade the Monarc version to the latest one. The password setup option is available there for new and existed users.

Please follow the upgrade procedure available on the website: https://www.monarc.lu/documentation/technical-guide/#monarc-update.

**Important note**
The update can be tough in case if your version is too old, before 2.9.1, so you need to follow the update process to the 2.9.0 first:
https://www.monarc.lu/news/2019/11/25/monarc-291-released/#updating and check all the system requirements before. For the latest Monarc version you need to have at least php 7.4.
Before the upgrade procedure, we would also recommend to make a DB backup and Monarc project root folder copy to be able to rollback in case if something is wrong. 
The ideal solution is to setup a new Monarc project in parallel, configure it to work with the existing DB (make a copy or dump before) and switch the Apache virtual host to work with the fresh instance.

**Reply by Pyr0technicien** [reply]
Thanks, unfortunatly we have to many duplicated keys error to perform the upgrade this is not an option for our customer

**Reply by ruslanbaidan** [reply]
You can comment out the following lines in your:
vendor/monarc/frontoffice/migrations/db/20210517000000_add_unique_code_constraint.php
https://github.com/monarc-project/zm-client/blob/master/migrations/db/20210517000000_add_unique_code_constraint.php#L10-L12
and
vendor/monarc/core/migrations/db/20210517000000_add_unique_code_constraint.php
https://github.com/monarc-project/zm-core/blob/master/migrations/db/20210517000000_add_unique_code_constraint.php#L10-L12
After execute the update process again.

**Comment by cedricbonhomme** [comment, marked as answer]
You can also change the email of the default user, [here](https://github.com/monarc-project/zm-client/blob/b1c9fb7e546bc5cfea624dd4d8eb1464378e7e25/migrations/seeds/AdminUserInit.php#L29=).

Then, execute again the script which creates the initial user:

```bash
$ php ./bin/phinx seed:run -c ./module/Monarc/FrontOffice/migrations/phinx.php
```

But the best is really to update your MONARC instance.

What is the current version of your MONARC instance ?

**Reply by Pyr0technicien** [reply]
Thanks that was indeed the way to do it.

This customer want to change platform so we arent gonna spend time upgrading but I just realised it was an even older version than our monarc. I'm still wondering how they managed to create these 5 users without smtp.

Thanks anyway, fast replay as usual !
